### ISSUE TITLE
pageserver: persist layer access time across restarts

### ISSUE THREAD

**Issue by problame** [neondatabase/neon#3727]
Currently, Pageserver does not persist the layer access stats, meaning we lose them after Pageserver restart.

The last-access-threshold-based eviction (`LayerAccessThreshold`) which we have today, therefore, does no evictions until after
`pageserver_restart_time + threshold_value + evaluation_period`.

Disk-pressure-based eviction (issue TBD), which we planned today (2023-03-01), can fall back to random selection,  file size /  atime / some derivative of mtime.

We also need to consider aspects like **tenant relocation**: do we want to migrate these statistics or start over on the new pageserver?

This issue is not super urgent, but it will need to get done at some point. Hopefully, the design space contracts more over the next few weeks.

**Comment by koivunej** [comment]
Discussion
- my idea: just use filesystem atime
- @hlinnaka noticed that on attach we don't have atime
  - always when uploading `index_part.json` include the then latest access time within metadata

**Comment by shanyp** [comment]
> We also need to consider aspects like tenant relocation: do we want to migrate these statistics or start over on the new pageserver?

once a tenant is migrated and get_page requests get in , will these "migrated" statistics still hold ?

**Comment by LizardWizzard** [comment]
Yes, as long as workload is the same access pattern shouldn't change

**Comment by problame** [comment]
(Cleaning up stale my issues.)

Today the situation hasn't fundamentally changed (we initialize the LayerAccessStats with `SystemTime::now()`.

But we don't have any work on this planned for the next few months, so, closing as not planned.
